File: memo/StaticReferenceField.java
class StaticReferenceField {
    public static StaticReferenceField srf = new StaticReferenceField();

    public int x;

    public StaticReferenceField() {
        this.x = 51;
    }

    public static int main(String[] args) {
        return srf.x;
    }
}

File: tests/static_reference_field.c
#include "../main.h"

int main(int argc, char *argv[]) {
    char *classes[1] = {"StaticReferenceField.class"};
    int retval = run(classes, 1);

    if (retval == 51) {
        return 0;
    } else {
        fprintf(stderr, "expect %d but actual %d\n", 51, retval);
        return 1;
    }
}


